### DISCUSSION TITLE
cant change default model

### DISCUSSION THREAD

**Discussion by fabiendostie** [danielmiessler/Fabric#1172] [Q&A]
On windows 11
using the fabric.exe v1.4.116

**my problem:**
using fabric /d
i get the:
`[Default]

Enter the index the name of your default model (leave empty for 'claude-3-5-sonnet-20240620' or type 'reset' to remove the value):
51

Enter model context length (leave empty to skip):

DEFAULT_VENDOR: OpenAI
DEFAULT_MODEL: chatgpt-4o-latest`

but after it doesn't seem to switch to it.

` pbpaste | .\fabric --pattern summarize
POST "https://api.anthropic.com/v1/messages": 400 Bad Request ...`

**Also: I'd like to know when using the .exe version, how to use the gui? i dont know where is the /web folder since it'sa .exe file?**

Thanks

**Comment by bledburn** [comment]
There are a couple things to know, before your questions can begin to have answers. OpenAI and 4o-latest should not be using an Anthropic API endpoint, as indicated by:

POST "https://api.anthropic.com/v1/messages": 400 Bad Request ...

Anthropic endpoints are for Claude and all of Anthropic's other models. Your base endpoint should be:

https://api.openai.com/v1

Which you need to enter when setting up OpenAI from the Vendor's List in the Fabric Setup menu. Then, as long as your OpenAI API key is properly entered, you should be good to go.

Next, the two GUI applications: the Pattern Mill (via the streamlit.py file), and the Web UI (via the web directory).

These are both located in the top of the Fabric repository. You can run the the Pattern Mill in a docker container, or more simply by using virtual environments, downloading the dependencies as listed in the requirements.txt file (also in the top of the repo) and then running the streamlit.py file with a python command. The Web UI, as I understand it, needs some things installed using the node package manager (npm). To do these things, it is easiest to clone the fabric repo down to your system. 

To be clear, the GUIs available for Fabric, are explicitly not a part of the Fabric CLI tool, they are separate projects as far as installation is concerned. I hope that makes sense.

**Reply by bledburn** [reply]
This speed tour of Miessler's Fabric AI might help things make more sense.

Watch here: https://youtu.be/g5lmDEQnq-A
